src/View/ViewContainer.vala

```diff
@@ -105,34 +138,51 @@ namespace Files.View {
 
         /* Initial location now set by Window.make_tab after connecting signals */
         public ViewContainer (View.Window win) {
-            window = win;
-            browser = new Browser ();
-
-            set_events (Gdk.EventMask.ENTER_NOTIFY_MASK | Gdk.EventMask.LEAVE_NOTIFY_MASK);
-            connect_signals ();
+            Object (
+                window: win
+            );
         }
 
-        ~ViewContainer () {
-            debug ("ViewContainer destruct");
-        }
-
-        private void connect_signals () {
+        construct {
+            browser = new Browser ();
             loading.connect ((loading) => {
                 is_loading = loading;
             });
 
-            button_press_event.connect (on_button_press_event);
-        }
-
-        private void connect_window_signals () {
-            if (window != null) {
-                window.folder_deleted.connect (on_folder_deleted);
-            }
+            var button_controller = new Gtk.GestureMultiPress (this) {
```

Comment:
I can't confirm that this works. Is there a reason we add this controller to the viewcontainer widget instead of adding it to the window and triggering `action_back` and `action_forward`? It seems like we'd want these mouse buttons to work when focus is anywhere in the window and not just in the viewcontainer as the keyboard shortcuts do

Reply:
I didnt think we would want this to work when e.g. the search bar was focused.  It is true that the keyboard shortcut does work in that case but I am not sure that it should. But if you disagree, I can associate this controller with the window instead.  I do not have an appropriate mouse to test this - I'll try to fake a test.

Reply:
Now moved to window.  I have added additional shortcuts for testing without a special mouse - these can be removed if not wanted.

Reply:
Ensure you have some paths in both the forward and backwards stacks for testing :smile: